Add Organisation and Project invitations (#560)
mongodb/terraform-provider-mongodbatlas: website/docs/r/org_invitation.html.markdown

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: org_invitation"
sidebar_current: "docs-mongodbatlas-organization-invitation"
description: |-

>     Provides a Atlas Organisation Invitation resource.

Reviewer: Just a reminder from data sources above: 
Please do a replace of all Organisation to Organization to match our Atlas doc and replace "a" before any org mention with an "an".

---

# mongodbatlas_org_invitation

> `mongodbatlas_org_invitation` provides a Organisation Invitation resource. This represents an invitation for an Atlas User within an Atlas Organisation.

Reviewer: an Organization

Each invitation for an Atlas user has a set of roles that provide access to an organization.

The roles that can be utilised can be found in the [MongoDB Documentation](https://docs.atlas.mongodb.com/reference/user-roles/#organization-roles), which map to:

* ORG_OWNER
* ORG_GROUP_CREATOR
* ORG_BILLING_ADMIN
* ORG_READ_ONLY
* ORG_MEMBER

-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.

> ~> **IMPORTANT:** All arguments including the password will be stored in the raw state as plain-text. [Read more about sensitive data in state.](https://www.terraform.io/docs/state/sensitive-data.html)

Reviewer: Bad copy/paste VII

## Example Usages

```hcl
resource "mongodbatlas_org_invitation" "test" {

>   username    = "test-acc-username"

Reviewer: Won't this always be an email address?

  org_id      = "<ORG-ID>"

>   roles       = [ "GROUP_DATA_ACCESS_READ_WRITE" ]

Reviewer: These aren't ORG roles, they should be one or more of 
    ORG_OWNER
    ORG_MEMBER
    ORG_GROUP_CREATOR
    ORG_BILLING_ADMIN
    ORG_READ_ONLY

Reviewer: Still a project role in example.

}
```

```hcl
resource "mongodbatlas_org_invitation" "test" {
  username    = "test-acc-username"
  org_id      = "<ORG-ID>"
  roles       = [ "GROUP_READ_ONLY", "GROUP_DATA_ACCESS_READ_ONLY" ]
}
```

## Argument Reference

> * `project_id` - (Required) The unique ID for the project to create the database user.

Reviewer: Bad copy/paste ... 8, I think I'm at 8

> * `username` - (Required) The Atlas user's email address.

Reviewer: I'd expand out to what we have in the docs (that's what the core DoU team normally does), e.g. for this: 
"Email address of the invited user. This is the address to which Atlas sends the invite.
If the user accepts the invitation, they log in to Atlas with this username."

* `roles` - (Required) 	List of user’s Atlas roles. The available options are:
  * ORG_OWNER
  * ORG_GROUP_CREATOR
  * ORG_BILLING_ADMIN
  * ORG_READ_ONLY
  * ORG_MEMBER

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

> * `id` - The database user's name.

Reviewer: also teamIds and inviterUsername if the comments above result in them being added.

* `created_at` - The date and time the invitation was created
* `expires_at` - The date and time that the invitation will expire
* `invitation_id` - The identify of the invitation in Atlas

## Import

> Organisations Invitations can be imported using organization ID and username (email address), in the format `org_id`-`username`, e.g.

Reviewer: I'd specify  "invitee username" to help clarify.

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: org_invitation"
sidebar_current: "docs-mongodbatlas-resource-organization-invitation"
description: |-
    Provides an Atlas Organization Invitation resource.
---

# mongodbatlas_org_invitation

> `mongodbatlas_org_invitation` provides an Organization Invitation resource. This represents an iIvitation for an Atlas User within an Atlas Organization.

Reviewer: Typo in the later sentence: This represents an **iIvitation** for an Atlas User within an Atlas Organization.

Author: Done (thanks for that)

> `mongodbatlas_org_invitation` provides an Organization Invitation resource. This represents an invitation for an Atlas User within an Atlas Organization.

Reviewer 2: ```suggestion
`mongodbatlas_org_invitation` invites a user to join an Atlas organization.
```

Author: Done

> Each Invitation for an Atlas User has a set of roles that provide access to an Organization.

Reviewer 2: lower case common nouns + edits for clarity
```suggestion
Each invitation for an Atlas user includes roles that Atlas grants the user when they accept the invitation.
```

Author: Done

> The roles that can be utilized can be found in the [MongoDB Documentation](https://docs.atlas.mongodb.com/reference/user-roles/#organization-roles), which map to:

Reviewer 2: try to remove passive voice
```suggestion
The [MongoDB Documentation](https://docs.atlas.mongodb.com/reference/user-roles/#organization-roles) describes the roles a user can have, which map to:
```

Author: Done

* ORG_OWNER
* ORG_GROUP_CREATOR
* ORG_BILLING_ADMIN
* ORG_READ_ONLY
* ORG_MEMBER

## Example Usages

```hcl
resource "mongodbatlas_org_invitation" "test0" {
  username    = "test0-acc-username"
  org_id      = "<ORG-ID>"
  roles       = [ "ORG_OWNER" ]
}
```

```hcl
resource "mongodbatlas_org_invitation" "test0" {
  username    = "test0-acc-username"
  org_id      = "<ORG-ID>"
  roles       = [ "ORG_MEMBER", "ORG_BILLING_ADMIN" ]
}
```

```hcl
resource "mongodbatlas_org_invitation" "test1" {
  username    = "test1-acc-username"
  org_id      = "<ORG-ID>"
  teams_ids   = [ "<TEAM-0-ID>", "<TEAM-1-ID>" ]
  roles       = [ "ORG_MEMBER" ]
}
```

## Argument Reference

* `org_id` - (Required) The unique ID for the Organization to create the Invitation.
* `username` - (Required) The Atlas User's email address.
* `teams_ids` - (Optional) An array of team identifiers that the Atlas User has been invited to join

> * `roles` - (Required) 	List of User’s roles within the Atlas Organization. The available options are:

Reviewer 2: ```suggestion
* `org_id` - (Required) Unique 24-hexadecimal digit string that identifies the organization to which you want to invite a user.
* `username` - (Required) Email address of the invited user. This is the address to which Atlas sends the invite. If the user accepts the invitation, they log in to Atlas with this username.
* `teams_ids` - (Optional) An array of unique 24-hexadecimal digit strings that identify the teams that the user was invited to join.
* `roles` - (Required) Atlas roles to assign to the invited user. If the user accepts the invitation, Atlas assigns these roles to them. The following options are available:
```

Author: Done

  * ORG_OWNER
  * ORG_GROUP_CREATOR
  * ORG_BILLING_ADMIN
  * ORG_READ_ONLY
  * ORG_MEMBER

## Attributes Reference

> In addition to all arguments above, the following attributes are exported:

Reviewer 2: ```suggestion
In addition to the arguments, this resource exports the following attributes:
```

Author: Done

* `id` - Autogenerated Unique ID for this resource.
* `created_at` - The date and time the Invitation was created
* `expires_at` - The date and time that the Invitation will expire
* `invitation_id` - The identity of the Invitation in Atlas

> * `inviter_username` - The username of the person creating the Invitation

Reviewer 2: ```suggestion
* `id` - Autogenerated unique string that identifies this resource.
* `created_at` - Timestamp in ISO 8601 date and time format in UTC when Atlas sent the invitation.
* `expires_at` - Timestamp in ISO 8601 date and time format in UTC when the invitation expires. Users have 30 days to accept an invitation.
* `invitation_id` - Unique 24-hexadecimal digit string that identifies the invitation in Atlas.
* `inviter_username` - Atlas user who invited `username` to the organization.
```

Author: Done

## Import

> Organisations Invitations can be imported using organization identifier and username (email address), in the format `org_id`-`username`, e.g.

Reviewer 2: ```suggestion
Import a user's invitation to an organization by separating the `org_id` and the `username` with a hyphen:
```

Author: Done
